pkg/cmd/package/upload/upload.go

```diff
@@ -0,0 +1,182 @@
+package upload
+
+import (
+	"errors"
+	"fmt"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/constants/annotations"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/util"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/newclient"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/packages"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// TODO Outstanding questions for package upload:
+// - What should the response be for outputformat basic and json?
+// - Basic output: I have "Successfully uploaded package x-1.0.0.zip" when the package was created/overwritten, and "Successfully processed package x-1.0.0.zip" when it was ignored. Is this right?
+// - If we specify multiple packages and one of them fails, does it keep going or abort on first error?
+// - the package upload response from the server has a load of stuff in it. Is there any value in doing anything with this?
```

Comment:
Not at the moment. But perhaps in the future?

Reply:
Cool, thanks for that. I'll remove this TODO comment and add another one at the callsite explaining that while `UploadPackageResponse` has a lot of data in it, the CLI deliberately doesn't use any at this time